napari/plugins/npe2api.py

```diff
@@ -71,22 +72,38 @@ def conda_map() -> Dict[PyPIname, Optional[str]]:
         return json.load(resp)
 
 
-def iter_napari_plugin_info() -> Iterator[Tuple[PackageMetadata, bool]]:
+def iter_napari_plugin_info() -> Iterator[
+    Tuple[PackageMetadata, bool, Dict[str, str]]
+]:
     """Iterator of tuples of ProjectInfo, Conda availability for all napari plugins."""
     with ThreadPoolExecutor() as executor:
-        data = executor.submit(pypi_plugin_summaries)
+        data = executor.submit(plugin_summaries)
         _conda = executor.submit(conda_map)
 
     conda = _conda.result()
-    for info in data.result():
+    for i, info in enumerate(data.result()):
+        # Sleep every 2 items for 150 ms to avoid hanging the UI
+        if i % 2 == 0:
+            time.sleep(0.150)
```

Comment:
What is the problem here? iteration over `data.result()`?

Reply:
Yes, the UI was hanging a little bit while iterating over `data.result()` and adding in a short pause helped.

Reply:
But the data result is already known, or is it really iterable with waiting on the network? Maybe we should change the approach and do not use yielding from another but add all elements in one iteration over the list?

Reply:
@Czaki can you expand a bit on your suggestion, I'm not sure I understand it? I agree that this sleep here seems kinda hacky and it would be nice to avoid. Seems to me like we should have a per plugin https request rather than one that gets all plugin summaries - that way we could yield after a very small request. I don't really get how it's working now though.